tty/pc-tty/ttypc_vt.c

```diff
@@ -0,0 +1,737 @@
+				break;
+
+			case '?':   /* ESC [ ? family */
+				vt->escst = ESC_CSIQM;
+				break;
+
+			case '@':   /* Insert char */
+				_ttypc_vtf_ic(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case '"':   /* Select char attribute */
+				vt->escst = ESC_SCA;
+				break;
+
+			case '!':   /* Soft terminal reset */
+				vt->escst = ESC_STR;
+				break;
+
+			case 'A':   /* Cursor up */
+				_ttypc_vtf_cuu(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'B':   /* Cursor down */
+				_ttypc_vtf_cud(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'C':   /* Cursor forward */
+				_ttypc_vtf_cuf(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'D':   /* Cursor backward */
+				_ttypc_vtf_cub(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'H':   /* Direct cursor addressing */
+				_ttypc_vtf_curadr(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'J':   /* Erase screen */
+				_ttypc_vtf_clreos(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'K':   /* Erase line */
+				_ttypc_vtf_clreol(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'L':   /* Insert line */
+				_ttypc_vtf_il(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'M':   /* Delete line */
+				_ttypc_vtf_dl(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'P':   /* Delete character */
+				_ttypc_vtf_dch(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'S':   /* Scroll up */
+				_ttypc_vtf_su(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'T':   /* Scroll down */
+				_ttypc_vtf_sd(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'X':   /* Erase character */
+				_ttypc_vtf_ech(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'c':   /* Device attributes */
+				_ttypc_vtf_da(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'f':   /* Direct cursor addressing */
+				_ttypc_vtf_curadr(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'g':   /* Clear tabs */
+				_ttypc_vtf_clrtab(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'h':   /* Set ANSI modes */
+				_ttypc_vtf_setansi(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'i':   /* Media copy */
+				_ttypc_vtf_mc(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'l':   /* Reset ANSI modes */
+				_ttypc_vtf_resetansi(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'm':   /* Select graphic rendition */
+				_ttypc_vtf_sgr(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'n':   /* Reports */
+				_ttypc_vtf_dsr(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'r':   /* Set scrolling region */
+				_ttypc_vtf_stbm(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'x':   /* Request/report parameters */
+				_ttypc_vtf_reqtparm(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			case 'y':   /* Invoke selftest(s) */
+				_ttypc_vtf_tst(vt);
+				vt->escst = ESC_INIT;
+				break;
+
+			default:
+				vt->escst = ESC_INIT;
+				break;
+			}
+			break;
+
+		case ESC_CSIQM: /* DEC private modes */
+			switch(c)
```

Comment:
missing space between switch and (c), { shouldn't be in the new line

Reply:
Fixed.